sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.clientTelemetry;
+
+import com.azure.cosmos.ConnectionMode;
+import com.azure.cosmos.implementation.Configs;
+import com.azure.cosmos.implementation.GlobalEndpointManager;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.cpu.CpuMemoryMonitor;
+import com.azure.cosmos.implementation.http.HttpClient;
+import com.azure.cosmos.implementation.http.HttpHeaders;
+import com.azure.cosmos.implementation.http.HttpRequest;
+import com.azure.cosmos.implementation.http.HttpResponse;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.netty.handler.codec.http.HttpMethod;
+import org.HdrHistogram.ConcurrentDoubleHistogram;
+import org.HdrHistogram.DoubleHistogram;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class ClientTelemetry {
+    public final static int REQUEST_LATENCY_MAX = 300000000;
+    public final static int REQUEST_LATENCY_SUCCESS_PRECISION = 4;
+    public final static int REQUEST_LATENCY_FAILURE_PRECISION = 2;
+    public final static String REQUEST_LATENCY_NAME = "RequestLatency";
+    public final static String REQUEST_LATENCY_UNIT = "MicroSec";
+
+    public final static int REQUEST_CHARGE_MAX = 10000;
+    public final static int REQUEST_CHARGE_PRECISION = 2;
+    public final static String REQUEST_CHARGE_NAME = "RequestCharge";
+    public final static String REQUEST_CHARGE_UNIT = "RU";
+
+    public final static int CPU_MAX = 100;
+    public final static int CPU_PRECISION = 2;
+    private final static String CPU_NAME = "CPU";
+    private final static String CPU_UNIT = "Percentage";
+
+    public final static int MEMORY_MAX = 102400;
+    public final static int MEMORY_PRECISION = 2;
+    private final static String MEMORY_NAME = "MemoryRemaining";
+    private final static String MEMORY_UNIT = "MB";
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private ClientTelemetryInfo clientTelemetryInfo;
+    private HttpClient httpClient;
+    private final ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
ideally we should use daemon threads otherwise if executor is not shutdown correctly JVM will not terminate.
Take a look at how rntbd sets up its daemon IO threads.

Reply:
done